### PR TITLE
Quadric

### PR DESCRIPTION
**Pull request by ScandalousMan**
Hello,

You can test the quadrics with adrien-04.json

quadrics include :
- ellipsoids
- hyperboloids
- paraboloids
- paraboloic cylinders

I will be able as an enhancement to define automatically the nature of the quadric drawn. (NOT DONE TODAY)

### CODE REVIEW

File: srcs/closest.c [hunk 1]
```diff
@@ -27,6 +31,8 @@ t_object	*closest_object(t_param *param, double *from, double *to, t_path *path)
 						pt_translated(from, path->valid_x, path->valid_x);
 						if (param->tmp_d > 0 && is_inside_object(objs, path))
 						{
+						if (point_display(param))
```
Comment (Its-Alex) [inline srcs/closest.c]: indent error ?

File: srcs/closest.c [hunk 2]
```diff
@@ -47,6 +55,8 @@ t_object	*closest_object(t_param *param, double *from, double *to, t_path *path)
 				}
 				if (param->tmp_d > 0.0)
 				{
+						if (point_display(param))
```
Comment (Its-Alex) [inline srcs/closest.c]: ident error ?

File: srcs/constructor.c [hunk 3]
```diff
@@ -45,6 +45,87 @@ t_param			*pxl_infos_create(t_param *param)
...
+t_sdl				*copy_pxls(t_param *param)
+{
+	t_sdl *graph;
+	int 	count;
+
+	count = -1;
+	if (!(graph = (t_sdl*)malloc(sizeof(t_sdl))))
+		return (NULL);
+	ft_bzero((void*)graph->input, sizeof(char) * SDL_NUM_SCANCODES);
+	while (++count < SDL_NUM_SCANCODES)
```
Comment (Its-Alex) [inline srcs/constructor.c]: This is for reset all keys ?

File: srcs/constructor.c [hunk 4]
```diff
@@ -45,6 +45,87 @@ t_param			*pxl_infos_create(t_param *param)
...
+	int 	count;
+
+	count = 0;
+	while (count < NB_THREAD)
+	{
+		SDL_FreeSurface(param->graph->surfs[count]);
+		SDL_FreeSurface(param->graph->tmp_surfs[count]);
+		count++;
+	}
+	free(param->graph);
```
Comment (Its-Alex) [inline srcs/constructor.c]: Why did you free surface ? Program exit ?

File: srcs/display.c [hunk 5]
```diff
@@ -80,5 +114,5 @@ void	display_lights(t_param *param)
 
 int		point_display(t_param *param)
 {
-	return (param->i[0] == 67 && param->i[1] == 263); // && param->is_for_light
+	return (param->i[0] == 327 && param->i[1] == 270); //
```
Comment (Its-Alex) [inline srcs/display.c]: What are there numbers ?

File: srcs/main.c [hunk 6]
```diff
@@ -33,6 +33,7 @@ int		main(int ac, char **av)
 	if (!rt_parser(param, filename))
 		return (1);
 	sdl_init(param->graph);
+	launch_threads(param);
```
Comment (Its-Alex) [inline srcs/main.c]: Always refresh scene ?

File: srcs/main.c [hunk 7]
```diff
@@ -44,13 +45,14 @@ int		main(int ac, char **av)
...
 		{
-			lauch_threads(param);
 			sdl_draw(param->graph);
 			param->end = clock();//TODO delete
 			printf("Render %.5lf secondes...\n", (double)(param->end - param->start) / CLOCKS_PER_SEC);
 			param->refresh = 0;
 		}
 	}
 	sdl_quit(param->graph);
+	while (1)
```
Comment (Its-Alex) [inline srcs/main.c]: Maybe delete this on master branch ?

File: srcs/closest.c (hunk 1 above)
Comment (ScandalousMan) [reply srcs/closest.c]: yes, this debug log will be removed at the end of the new objects implementation so we don't delete and rewrite the same lines over and over. Not to be kept.

File: srcs/closest.c (hunk 2 above)
Comment (ScandalousMan) [reply srcs/closest.c]: yes, this debug log will be removed at the end of the new objects implementation so we don't delete and rewrite the same lines over and over. Not to be kept.

File: srcs/constructor.c (hunk 3 above)
Comment (ScandalousMan) [reply srcs/constructor.c]: I guess, I never wrote this part. I don't understand why it is considered as part of this PR

**Pull request merged**